docs/gaps/README.md

```diff
@@ -0,0 +1,190 @@
+  must ensure this is reflected in the living GAP document
+
+Changes to GAP documents are subject to approval from the same threshold of
+gittuf maintainers as all implementation changes. That is, if the implementation
+requires two maintainers to approve some change, the same threshold applies to
+changes to GAPs.
+
+## GAP Workflow
+
+A GAP may go through the following phases in its evolution from problem to
+implemented solution.
+
+### Discuss the problem
+
+A GAP solves specific problems that the gittuf design currently does not. Rather
+than directly approach the community with a GAP, it is a good idea to discuss
+with the maintainers and the broader community the problem itself. This can help
+confirm that the problem in fact exists (instead of being a misunderstanding of
+the gittuf design), has not already been explored in a previous GAP, and a
+solution ought to be part of gittuf (rather than another complementary system).
+This discussion may happen in any forum used by the gittuf community, though the
+repository's issue tracker is recommended for maximum visibility.
+
+### Propose the GAP
+
+After the maintainers agree that a GAP is required, one or more sponsors can
+author a draft of the GAP. To submit a GAP, one of the sponsors must open a pull
+request with the document to the gittuf repository. The GAP must follow the
+format specified in this document (or copy the template provided alongside this
+document), minus the number as that is assigned when the document is merged.
+
+### Merging a proposed GAP
+
+All proposed GAPs must be merged into the repository in a timely manner,
+regardless of their implementation status. This increases the visibility of each
+GAP, thus making it easier for other interested parties to discover the GAP,
+propose further changes, and contribute to the implementation of the GAP.
+
+The sponsors of a GAP may choose to withdraw their proposal or the maintainers
+may choose to reject the proposed changes after assessing the GAP. Even in these
+cases, the document must be merged into the repository with the status indicated
+in the document's metadata section. Note that a GAP must only be withdrawn or
+rejected on the basis of technical reasons (e.g., a better solution is proposed
+or a security issue is discovered a consequence of the proposal). The reasoning
+section of the GAP must capture these technical considerations.
+
+### Implementing a GAP
+
+The changes proposed in a GAP may be implemented via patches to the gittuf
+implementation. The changes to the implementation need not be submitted by the
+sponsors of the GAP. When the gittuf maintainers think a GAP has been
+implemented, they can propose an update to the document reflecting this. The
+sponsors of a GAP may also propose marking a GAP as implemented, which is
+subject to approval from the gittuf maintainers.
+
+### Removing a GAP's implementation
+
+If a GAP's implementation is removed, then the GAP must be updated to indicate
```

Comment:
It would be nice to give a rationale as to why a GAP's implementation may be removed (maybe give an example).

Reply:
Added some example scenarios. I think this is likely to fall into two categories:
a) the feature is a security nightmare and we are willing to take the backwards compatibility hit to remove it
b) the feature is made redundant due to other GAPs or more fundamental changes to gittuf's design